### PR TITLE
Api 17683/add a packagejson script for building test code

### PR DESCRIPTION
**Pull request by stevelong00**
API-17683/Add-a-packagejson-script-for-building-test-code

- Added a script and updated configs to support building test code
- Added comments to tsconfig

### CODE REVIEW

File: jest.config.js [hunk 1]
```diff
@@ -12,9 +12,10 @@ module.exports = {
       statements: 80,
     },
   },
-  coveragePathIgnorePatterns: ['/node_modules/', 'base.entity.ts', '.model.ts'],
-  testPathIgnorePatterns: ['/node_modules/', '/dist/'],
+  coveragePathIgnorePatterns: ['/node_modules/', 'base.entity.ts', '.model.ts', '/dist-test/'],
+  testPathIgnorePatterns: ['/node_modules/', '/dist/', '/dist-test/'],
   testEnvironment: 'node',
   testRegex: '(/__tests__/.*|(\\.|/)(test|integration))\\.[jt]sx?$',
+  testTimeout: 20000, // milliseconds
```
Comment (mheusgit) [inline jest.config.js:19]: Looks like duplicate work to API-16584

Comment (stevelong00) [reply jest.config.js]: Timeout change removed

**Review by peyton-lucas — COMMENTED** [review]
LGTM!

**Review by mheusgit — APPROVED** [review]
Looks good

**Review by dsteeve — APPROVED** [review]
good to go!

**Pull request merged**